micro-rdk/src/common/app_client.rs

```diff
@@ -296,39 +296,6 @@ impl AppClient {
             None
         };
 
-        #[cfg(feature = "esp32")]
```

Comment:
I'm not sure what the benefit is of having both ntp at startup (setting right after connecting) and this during `get_app_config`. 

Additionally, the sntp library handles periodic setting (seen below in sdkconfig)

Reply:
I could see this being retained as a fallback in case ntp isn't working. If the NTP stuff set `CLOCK_SET` when the time was synced, if you got here and it still wasn't synced you could use the app time.